homeassistant/components/nobo_hub/climate.py

```diff
@@ -194,25 +194,28 @@ def _read_state(self) -> None:
         self._attr_current_temperature = (
             None if current_temperature is None else float(current_temperature)
         )
-        if ClimateEntityFeature.TARGET_TEMPERATURE in self._attr_supported_features:
-            self._attr_target_temperature_high = None
-            self._attr_target_temperature_low = None
-            if self._supports_comfort:
-                temp_attr = ATTR_TEMP_COMFORT_C
-            else:
-                temp_attr = ATTR_TEMP_ECO_C
-            self._attr_target_temperature = int(self._nobo.zones[self._id][temp_attr])
-        elif (
-            ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
-            in self._attr_supported_features
-        ):
-            self._attr_target_temperature_high = int(
-                self._nobo.zones[self._id][ATTR_TEMP_COMFORT_C]
-            )
-            self._attr_target_temperature_low = int(
-                self._nobo.zones[self._id][ATTR_TEMP_ECO_C]
-            )
-            self._attr_target_temperature = None
+        if self._attr_supported_features:
+            if ClimateEntityFeature.TARGET_TEMPERATURE in self._attr_supported_features:
```

Comment:
I see most tests on feature flags use the form `self._attr_supported_features & ClimateEntityFeature.TARGET_TEMPERATURE`

I don't know what is the preferred way though.